dotnet/src/Extensions/Planning.StepwisePlanner/StepwisePlanner.cs

```diff
@@ -33,39 +39,43 @@ public class StepwisePlanner : IStepwisePlanner
     /// </summary>
     /// <param name="kernel">The semantic kernel instance.</param>
     /// <param name="config">Optional configuration object</param>
-    /// <param name="prompt">Optional prompt override</param>
-    /// <param name="promptUserConfig">Optional prompt config override</param>
     public StepwisePlanner(
         IKernel kernel,
-        StepwisePlannerConfig? config = null,
-        string? prompt = null,
-        PromptTemplateConfig? promptUserConfig = null)
+        StepwisePlannerConfig? config = null)
     {
         Verify.NotNull(kernel);
         this._kernel = kernel;
 
         this.Config = config ?? new();
         this.Config.ExcludedSkills.Add(RestrictedSkillName);
 
-        var promptConfig = promptUserConfig ?? new PromptTemplateConfig();
-        var promptTemplate = prompt ?? EmbeddedResource.Read("Skills.StepwiseStep.skprompt.txt");
+        var prompt = this.Config.GetPromptTemplate is not null ? this.Config.GetPromptTemplate() : null;
+        this._promptTemplate = prompt ?? EmbeddedResource.Read("Skills.StepwiseStep.skprompt.txt");
+        this._promptConfig = this.Config.PromptUserConfig ?? new PromptTemplateConfig();
 
-        if (promptUserConfig == null)
+        this._manualTemplate = EmbeddedResource.Read("Skills.RenderFunctionManual.skprompt.txt");
+        this._questionTemplate = EmbeddedResource.Read("Skills.RenderQuestion.skprompt.txt");
+
+        if (this.Config.PromptUserConfig == null)
         {
             string promptConfigString = EmbeddedResource.Read("Skills.StepwiseStep.config.json");
             if (!string.IsNullOrEmpty(promptConfigString))
             {
-                promptConfig = PromptTemplateConfig.FromJson(promptConfigString);
+                this._promptConfig = PromptTemplateConfig.FromJson(promptConfigString);
             }
         }
+        else
+        {
+            this._promptConfig = this.Config.PromptUserConfig;
```

Comment:
this assignment isn't needed since it's already assigned in line 54, or line 54 isn't needed